From acea74b051d4f3d897f0999bf11edf86f72028b6 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mkbestas@lineageos.org>
Date: Thu, 1 Feb 2018 01:17:41 +0000
Subject: [PATCH] Revert "Revert "audio: add support for extended audio
 features""

* msm8974 audio HAL needs this for now

This reverts commit 2748aea48206ac9159bbb95d5e2d5bcd6761c956.

Change-Id: I5594d67d6605912537c5871381caf5e36a4cd379
---
 audio/include/system/audio-base.h | 5 +++++
 audio/include/system/audio.h      | 7 +++++++
 2 files changed, 12 insertions(+)

diff --git a/audio/include/system/audio-base.h b/audio/include/system/audio-base.h
index 441226d..f8a6f3a 100644
--- a/audio/include/system/audio-base.h
+++ b/audio/include/system/audio-base.h
@@ -99,6 +99,7 @@ typedef enum {
     AUDIO_FORMAT_APTX_HD = 553648128u, // 0x21000000UL
     AUDIO_FORMAT_AC4 = 570425344u, // 0x22000000UL
     AUDIO_FORMAT_LDAC = 587202560u, // 0x23000000UL
+    AUDIO_FORMAT_PCM_OFFLOAD = 603979776u, // 0x24000000UL
     AUDIO_FORMAT_MAIN_MASK = 4278190080u, // 0xFF000000UL
     AUDIO_FORMAT_SUB_MASK = 16777215u, // 0x00FFFFFFUL
     AUDIO_FORMAT_PCM_SUB_16_BIT = 1u, // 0x1
@@ -146,6 +147,8 @@ typedef enum {
     AUDIO_FORMAT_AAC_ADTS_LD = 503316608u, // (AAC_ADTS | AAC_SUB_LD)
     AUDIO_FORMAT_AAC_ADTS_HE_V2 = 503316736u, // (AAC_ADTS | AAC_SUB_HE_V2)
     AUDIO_FORMAT_AAC_ADTS_ELD = 503316992u, // (AAC_ADTS | AAC_SUB_ELD)
+    AUDIO_FORMAT_PCM_16_BIT_OFFLOAD = 603979777u, // (PCM_OFFLOAD | PCM_SUB_16_BIT),
+    AUDIO_FORMAT_PCM_24_BIT_OFFLOAD = 603979780u, // (PCM_OFFLOAD | PCM_SUB_8_24_BIT),
 } audio_format_t;
 
 enum {
@@ -222,6 +225,7 @@ enum {
     AUDIO_CHANNEL_INDEX_MASK_6 = 2147483711u, // (INDEX_HDR | ((1 << 6) - 1))
     AUDIO_CHANNEL_INDEX_MASK_7 = 2147483775u, // (INDEX_HDR | ((1 << 7) - 1))
     AUDIO_CHANNEL_INDEX_MASK_8 = 2147483903u, // (INDEX_HDR | ((1 << 8) - 1))
+    AUDIO_CHANNEL_IN_5POINT1 = 252u, // IN_LEFT | IN_RIGHT | IN_FRONT | IN_BACK | IN_LEFT_PROCESSED | IN_RIGHT_PROCESSED
 };
 
 enum {
@@ -324,6 +328,7 @@ typedef enum {
     AUDIO_OUTPUT_FLAG_DIRECT_PCM = 8192, // 0x2000
     AUDIO_OUTPUT_FLAG_MMAP_NOIRQ = 16384, // 0x4000
     AUDIO_OUTPUT_FLAG_VOIP_RX = 32768, // 0x8000
+    AUDIO_OUTPUT_FLAG_COMPRESS_PASSTHROUGH = 65536, // 0x10000
 } audio_output_flags_t;
 
 typedef enum {
diff --git a/audio/include/system/audio.h b/audio/include/system/audio.h
index e683d60..82f7443 100644
--- a/audio/include/system/audio.h
+++ b/audio/include/system/audio.h
@@ -853,6 +853,11 @@ static inline bool audio_is_valid_format(audio_format_t format)
     case AUDIO_FORMAT_AC4:
     case AUDIO_FORMAT_LDAC:
         return true;
+    case AUDIO_FORMAT_PCM_OFFLOAD:
+        if (format != AUDIO_FORMAT_PCM_16_BIT_OFFLOAD &&
+                format != AUDIO_FORMAT_PCM_24_BIT_OFFLOAD)
+            return false;
+        return true;
     default:
         return false;
     }
@@ -898,6 +903,7 @@ static inline size_t audio_bytes_per_sample(audio_format_t format)
     switch (format) {
     case AUDIO_FORMAT_PCM_32_BIT:
     case AUDIO_FORMAT_PCM_8_24_BIT:
+    case AUDIO_FORMAT_PCM_24_BIT_OFFLOAD:
         size = sizeof(int32_t);
         break;
     case AUDIO_FORMAT_PCM_24_BIT_PACKED:
@@ -905,6 +911,7 @@ static inline size_t audio_bytes_per_sample(audio_format_t format)
         break;
     case AUDIO_FORMAT_PCM_16_BIT:
     case AUDIO_FORMAT_IEC61937:
+    case AUDIO_FORMAT_PCM_16_BIT_OFFLOAD:
         size = sizeof(int16_t);
         break;
     case AUDIO_FORMAT_PCM_8_BIT:
-- 
2.7.4

